.github/workflows/test.yml

```diff
@@ -53,7 +54,7 @@ jobs:
           # Run unit tests
           dotnet test --no-build -c ${{ matrix.configuration }} ${SOLUTION}
           # Run lit test suite
-          lit -v --timeout=120 -D configuration=${{ matrix.configuration }} Test
+          ./Test/test-with-param.sh ${{ matrix.boogie_param }} -v --timeout=120 -D configuration=${{ matrix.configuration }}
```

Comment:
I'm pretty sure you can achieve the same thing by doing the following, instead of adding the extra scripting layer on top:

1. Define an extra parameter for lit that can be passed on the command line, something like `prover-batch-mode`. See https://github.com/dafny-lang/dafny/blob/master/Test/lit.site.cfg#L134 for how the Dafny config adds a `dafny_params` parameter (although in this case a boolean switch probably makes more sense).
2. Add a custom lit "feature" if and only if batch mode is turned on. Dafny uses this to communicate what platform the tests are running on: https://github.com/dafny-lang/dafny/blob/master/Test/lit.site.cfg#L154-L161
3. Use the `UNSUPPORTED` lit command prefix to skip tests that don't support the "feature" above.

Reply:
Oh, nice! I'll give that a try.